### ISSUE TITLE
Ideas on future changes to the flow and structure of RP-0

### ISSUE THREAD

**Issue by stratochief66** [KSP-RO/RP-0#510]
Just stealing notes from @NathanKell on the IRC, since they are worth having permanently referable:

<NathanKell|Busy> Here are my current thoughts:
1. More smaller nodes early.
This lets us spread some of those nodes where there are multiple engine configs. That also means you wait a month or two for a node, not a year.
2. Launch costs as part of a larger cost rework.
3. Tweak the gameplay mode defaults.
We should balance around, say, 80% rewards, I think. Because right now Hard feels like "normal difficulty"
and even Normal should default to having entrycosts on.
4. Disable/ignore fund loss slider. It's a mess.

**Comment by stratochief66** [comment]
<schnobs`> And for entry costs, we need a solution so it doesn't punish the use of part packs (opting for the "true" Atlas rather than a proc-tank knockoff).
<NathanKell|Busy> schnobs: Yes. Proc parts need unlock costs.

**Comment by stratochief66** [comment]
Agathorn @jwvanderbeck said some good things too, but not easily copy-pastable. Basically, an argument that TF presence should reduce entryCosts that he and I seem to have interest in.

**Comment by Schnobs** [comment]
About making a tech tree: could one just lift the tree from the ConfigCache and use that as a starting point?
More generally, I wonder if it's possible to just make a tree from scratch rather than apply a thousand patches to the stock tree.

**Comment by SirKeplan** [comment]
@Schnobs i think the plan is still to make a new tree roughly like this one #224

**Comment by PhineasFreak** [comment]
@Schnobs i did try and recreate from scratch (no CTT or RP-0 CTT patches) the tech tree that @SirKeplan mentioned above. Got it till the Tier 5 (i think): https://github.com/PhineasFreak/PicBin/commit/6480810ecff8e502f168a3c115578a9112bc80af

I always wondered why the tech tree was never finished in the first place.

**Comment by batuhantasli** [comment]
@PhineasFreak are you still working on that tech tree? I agree with you by the way, first of all we should complete the tech tree before we start placing/adjusting parts for 1.1 release. I might help on that as well.

**Comment by Schnobs** [comment]
@PhineasFreak also, could you post the source you have so far? I'm still looking for any kind of template that will work.

**Comment by PhineasFreak** [comment]
@batuhantasli yes but i stopped due to real life commitments (placing the nodes correctly takes a lot of time and KSP restarts unfortunately). But, as @Schnobs mentioned, i can easily provide the current WIP patch for everyone to buid upon it:

https://github.com/PhineasFreak/ROExpansion/blob/master/ROExpansion/ROE_Tech_Tree.cfg

Ignore the part config lines at the end of the file (not part of the tech tree patch).

**Comment by NathanKell** [comment]
Now that I'm less crunchy I'll look through this soon.

**Comment by Zarbizaure** [comment]
also, will the new RP0 keep its dependencies? We should definitely use RO-made mods instead of stock mods adapted into RO. Especially about SXT (SXT is cool, but not with RO imho): yes, you have some new engines, but a lot of tanks; and if you deleted the squad parts you don't need, almost all of the SXT parts will be borked (well, at least Bonanza/Junker/... in my case.)

Why don't we rely on others mods that will have a better integration with RO? SSTU, maybe?
(sorry Lack if you hear me, I love your mod but it's just too messy with RO....)

**Comment by Schnobs** [comment]
@PhineasFreak: I've had some kind of success with ripping the current RP-0 tree from ModuleManager.ConfigCache and editing that. I has most nodes in the right place, so it's mostly a question of renaming some and adjusting dependencies.

In the hopes of making it easier I started splitting it into several files along it's length: all hydrolox in one file, all rockets in another... the idea is that finishing one of these files is a task that can be completed in one sitting. Also, structuring the work. Then again, I spent most of an evening on splitting and still am not done.
--
One thing that bugs me is stuff like this:
```
			id = advRocketry
			title = Mature Orbital Rocketry
```
I'd like ID's to be a) consistent and b) not at odds with the in-game title. @NathanKell, would you prefer the IDs to be derived from the title ("MatureOrbitalRockerty"), or generic IDs ("rocketry3" or perhaps "rocketry1965")?

**Comment by NathanKell** [comment]
All implemented (apart from entry costs for proc parts which we hope to solve with @saabstory88 's new tanks).
